### Задание 1.1

Импортируйте библиотеку Numpy и дайте ей псевдоним np.
Создайте массив Numpy под названием a размером 5x2, то есть состоящий из 5 строк и 2 столбцов.
Первый столбец должен содержать числа 1, 2, 3, 3, 1, а второй - числа 6, 8, 11, 10, 7. Будем считать,
что каждый столбец - это признак, а строка - наблюдение. Затем найдите среднее значение по
каждому признаку, используя метод mean массива Numpy. Результат запишите в массив mean_a, в
нем должно быть 2 элемента.

In [27]:
import numpy as np

In [28]:
a = np.array([[1, 6],
             [2, 8],
             [3, 11],
             [3, 10],
             [1, 7]])
a

array([[ 1,  6],
       [ 2,  8],
       [ 3, 11],
       [ 3, 10],
       [ 1,  7]])

In [29]:
mean_a = a.mean(axis = 0)
mean_a

array([2. , 8.4])

### Задание 1.2

Вычислите массив a_centered, отняв от значений массива “а” средние значения соответствующих признаков, содержащиеся в массиве mean_a. Вычисление должно производиться в одно действие. Получившийся массив должен иметь размер 5x2.

In [30]:
a_centered = a - mean_a
a_centered


array([[-1. , -2.4],
       [ 0. , -0.4],
       [ 1. ,  2.6],
       [ 1. ,  1.6],
       [-1. , -1.4]])

### Задание 1.3

Найдите скалярное произведение столбцов массива a_centered. В результате должна получиться
величина a_centered_sp. Затем поделите a_centered_sp на N-1, где N - число наблюдений.

In [31]:
a_centered_sp = a_centered[0]
for i in range(len(a_centered)-1):
    a_centered_sp = np.dot(a_centered_sp, a_centered[i+1])
a_centered_sp

array([-4.9536 , -6.93504])

т.к. по условию задания 1.1 каждая строка - это наблюдение, то N = 5

In [32]:
a_centered_sp/(5-1)

array([-1.2384 , -1.73376])

### Задание 1.4**

Проверьте получившееся число, вычислив ковариацию еще одним способом - с
помощью функции np.cov